RecoLocalCalo/EcalRecAlgos/interface/DeclsForKernels.h

```diff
@@ -38,201 +37,363 @@ class EcalSamplesCorrelation;
-        cudaCheck(cudaFree(activeBXs));
-        cudaCheck(cudaFree(acState));
+        cudaCheck( cudaFree(noisecov) );
+        cudaCheck( cudaFree(pulse_matrix) );
+        cudaCheck( cudaFree(activeBXs) );
+        cudaCheck( cudaFree(acState) );
 
-        cudaCheck(cudaFree(hasSwitchToGain6));
-        cudaCheck(cudaFree(hasSwitchToGain1));
-        cudaCheck(cudaFree(isSaturated));
+        cudaCheck( cudaFree(hasSwitchToGain6) );
+        cudaCheck( cudaFree(hasSwitchToGain1) );
+        cudaCheck( cudaFree(isSaturated) );
 
         if (configParameters.shouldRunTimingComputation) {
-          cudaCheck(cudaFree(sample_values));
-          cudaCheck(cudaFree(sample_value_errors));
-          cudaCheck(cudaFree(useless_sample_values));
-          cudaCheck(cudaFree(chi2sNullHypot));
-          cudaCheck(cudaFree(sum0sNullHypot));
-          cudaCheck(cudaFree(sumAAsNullHypot));
-          cudaCheck(cudaFree(pedestal_nums));
-
-          cudaCheck(cudaFree(tMaxAlphaBetas));
-          cudaCheck(cudaFree(tMaxErrorAlphaBetas));
-          cudaCheck(cudaFree(accTimeMax));
-          cudaCheck(cudaFree(accTimeWgt));
-          cudaCheck(cudaFree(ampMaxAlphaBeta));
-          cudaCheck(cudaFree(ampMaxError));
-          cudaCheck(cudaFree(timeMax));
-          cudaCheck(cudaFree(timeError));
-          cudaCheck(cudaFree(tcState));
+            cudaCheck( cudaFree(sample_values) );
+            cudaCheck( cudaFree(sample_value_errors) );
+            cudaCheck( cudaFree(useless_sample_values) );
+            cudaCheck( cudaFree(chi2sNullHypot) );
+            cudaCheck( cudaFree(sum0sNullHypot) );
+            cudaCheck( cudaFree(sumAAsNullHypot) );
+            cudaCheck( cudaFree(pedestal_nums) );
+
+            cudaCheck( cudaFree(tMaxAlphaBetas) );
+            cudaCheck( cudaFree(tMaxErrorAlphaBetas) );
+            cudaCheck( cudaFree(accTimeMax) );
+            cudaCheck( cudaFree(accTimeWgt) );
+            cudaCheck( cudaFree(ampMaxAlphaBeta) );
+            cudaCheck( cudaFree(ampMaxError) );
+            cudaCheck( cudaFree(timeMax) );
+            cudaCheck( cudaFree(timeError) );
+            cudaCheck( cudaFree(tcState) );
         }
+    }
+};
+
+// const refs products to conditions
+struct ConditionsProducts {
+    EcalPedestalsGPU::Product const& pedestals;
+    EcalGainRatiosGPU::Product const& gainRatios;
+    EcalPulseShapesGPU::Product const& pulseShapes;
+    EcalPulseCovariancesGPU::Product const& pulseCovariances;
+    EcalSamplesCorrelationGPU::Product const& samplesCorrelation;
+    EcalTimeBiasCorrectionsGPU::Product const& timeBiasCorrections;
+    EcalTimeCalibConstantsGPU::Product const& timeCalibConstants;
+    EcalSampleMask const& sampleMask;
+    EcalTimeOffsetConstant const& timeOffsetConstant;
+    uint32_t offsetForHashes;
+};
+
+//*/
+
+struct xyz {
+    int x,y,z;
+};
+
+struct conf_data {
+    xyz threads;
+    bool runV1;
+    cudaStream_t cuStream;
+};
+
+}}        
+        
+        
+  
+// 
```

Comment:
Wouldn't it be better to have al headers at the top ?
Otherwise, you could split this into tow different files.

Reply:
done